z

```diff
@@ -0,0 +1,339 @@
+#   print_warning "Print a warning message."
+#
+print_warning() {
+    echo -e "${YELLOW}[WARN] ${1}${NC}" >&2
+}
+
+#
+# Description
+#   Prints help information on stdout.
+#
+# Usage Example
+#   print_help
+#
+print_help() {
+    cat << EOF
+Utility for building Nanvix.
+
+Usage:
+  ./z COMMAND [OPTIONS] [${OPT_NAME_SEPARATOR} BUILD_PARAMETERS]
+  ./z ${CMD_NAME_BUILD} [${OPT_NAME_WITH_DOCKER}] ["${OPT_NAME_SEPARATOR}" BUILD_PARAMETERS]
+  ./z ${CMD_NAME_CLEANUP} [${OPT_NAME_WITH_DOCKER}]
+  ./z ${CMD_NAME_HELP}
+
+Commands
+  build    Builds Nanvix.
+  cleanup  Cleans up build artifacts.
+  help     Prints the help information.
+
+Options
+  ${OPT_NAME_WITH_DOCKER}  Builds Nanvix using Docker instead of the native toolchain.
+EOF
+}
+
+#
+# Description
+#   Builds the Docker command to run the build process.
+#
+# Arguments
+#   $* - The build parameters to pass to the Docker command.
+#
+# Return
+#   A string containing the Docker command to run.
+#
+# Usage Example
+#   build_docker_command "all"
+#
+build_docker_command() {
+    local build_parameters="${*}"
+    local user_id
+    local group_id
+    user_id="$(id -u)"
+    group_id="$(id -g)"
+    echo "docker run -it --rm \
+        -v \"$(pwd):/mnt\" \
+        ${DOCKER_IMAGE_NAME} \
+        /bin/bash -l -c '\
+            cd /mnt ; \
+            git config --global --add safe.directory /mnt ; \
+            make TOOLCHAIN_DIR=${DOCKER_IMAGE_TOOLCHAIN_PATH} ${build_parameters} ; \
+            chown -R \"${user_id}:${group_id}\" .\
+        '"
+}
+
+#
+# Description
+#   Writes the last build command and options to the cache file.
+#   Each argument is written on a separate line.
+#
+# Arguments
+#   $@ - The full command line arguments.
+#
+# Usage Example
+#   write_cache "build" "--with-docker" "--" "all"
+#
+write_cache() {
+    printf '%s\n' "${@}" > "${Z_CACHE_FILE}"
+}
+
+#
+# Description
+#   Reads the last build command and options from the cache file.
+#   Sets LAST_BUILD_WITH_DOCKER based on whether the cache contains the Docker option.
+#
+# Usage Example
+#   read_cache
+#
+read_cache() {
+    if [[ -f "${Z_CACHE_FILE}" ]]; then
+        local cache_content
+        cache_content=$(cat "${Z_CACHE_FILE}")
+        if [[ "${cache_content}" == *"${OPT_NAME_WITH_DOCKER}"* ]]; then
+            LAST_BUILD_WITH_DOCKER=true
+        else
+            LAST_BUILD_WITH_DOCKER=false
+        fi
+    fi
+}
+
+#==================================================================================================
+# Main Script
+#==================================================================================================
+
+main() {
+    # Check if the script is running on a valid Git repository.
+    if ! git rev-parse --is-inside-work-tree &> /dev/null; then
+        print_error "This script must be run inside a Git repository."
+        exit 1
+    fi
+
+    # Check if the script is from the root of the repository.
+    if [[ ! -f "${REPO_ROOT_DIR}/z" ]]; then
```

Comment:
I think this is just checking that the `z` script is located in `${REPO_ROOT_DIR}/z` not that it is _run_ from ${REPO_ROOT_DIR}.

To test:

```
csegarra@milan1:~/git/nanvix/nanvix/src$ git rev-parse --show-toplevel
/home/csegarra/git/nanvix/nanvix
```